ktor-server/ktor-server-core/jvmAndNix/src/io/ktor/server/config/MergedApplicationConfig.kt

```diff
@@ -4,13 +4,30 @@
 
 package io.ktor.server.config
 
+@Deprecated("Use mergeWith/mergeWithFallback instead.")
+public fun List<ApplicationConfig>.merge(): ApplicationConfig {
+    require(isNotEmpty()) { "List of configs can not be empty" }
+    return foldRight(last()) { config, acc -> config.mergeWithFallback(acc) }
+}
+
 /**
  * Merge configuration combining all their keys.
- * If key is not found in one of the configs, search will continue in the next config in the list.
+ * If the key is in both configs - the value from Second config will be used.
```

Comment:
"If the key is in both `this` and [other] -- the value from [other] config will be used."

But @andreyaksenov can help better with phrasing.